File: mfp-class/begginer-class/pt2_cpp/beecrowd.cpp
// BeeCrowd

#include <iostream>
#include <vector>

using namespace std;

// Problema 1037
string result(float n) {
  if (n >= 0 && n <= 25)
    return "Intervalo [0,25]";
  else if (n > 25 && n <= 50)
    return "Intervalo (25,50]";
  else if (n > 50 && n <= 75)
    return "Intervalo (50,75]";
  else if (n > 75 && n <= 100)
    return "Intervalo (75,100]";

  return "Fora de intervalo";
}

void print_1037() {
  float number;
  cin >> number;

  cout << result(number) << endl;
}

// Problema 1059

void print_1059() {
  for (int i = 1; i <= 100; i++) {
    if (i % 2 == 0)
      cout << i << endl;
  }
}

// Problema 1176
void print_1176() {
  vector<int> fib;
  unsigned int a1 = 0;
  unsigned int a2 = 1;
  unsigned int auxiliar;

  fib.push_back(a1);
  fib.push_back(a2);

  for (int i = 2; i <= 60; i++) {
    auxiliar = a1 + a2;
    fib.push_back(auxiliar);

    a1 = a2;
    a2 = auxiliar;
  }

  int value, position;

  cin >> value;

  for (int i = 0; i < value; i++) {
    cin >> position;
    cout << "Fib(" << position << ") = " << fib[position] << endl;
  }
}

File: mfp-class/begginer-class/pt1_python/beecrowd.py
# BeeCrowd
# Problema 1037

number = float(input())

if number >= 0 and number <= 25:
    print("Intervalo [0,25]")
elif number > 25 and number <= 50:
    print("Intervalo (25,50]")
elif number > 50 and number <= 75:
    print("Intervalo (50,75]")
elif number > 75 and number <= 100:
    print("Intervalo (75,100]")
else:
    print("Fora de intervalo")


# Problema 1059
for i in range (1, 101):
  if(i % 2 == 0):
    print(i)

# Problema 1176
Fib = []
Fib.append(0)
Fib.append(1)

for i in range(2, 61):
  Fib.append(Fib[i-2] + Fib[i-1])

n = int(input())

for j in range(0, n):
  i = int(input())
  print("Fib(", i, ") = ", Fib[i], sep="")

# Problema 1142, 1175, 1240, 2003 - TODO